src/containers/SearchBar/index.js

```diff
@@ -24,19 +23,18 @@ export const SearchBarComponent = ({
    *
    * @param {String} searchInput
    */
-  const onSearchSubmit = searchInput => {
+  const onSearchSubmit = useCallback(searchInput => {
     onApplyFilter({});
-    onSetSearchQuery(searchInput);
     history.push('/manage/incidents');
-    onRequestIncidents();
-  };
+    onSearchIncidents(searchInput);
+  }, [history, onApplyFilter, onSearchIncidents]);
 
-  const onChange = value => {
+  const onChange = useCallback(value => {
     if (value === '') {
-      onSetSearchQuery('');
+      onSearchIncidents('');
```

Comment:
Will this fire off an empty search request right before a regular fetchIncidents?

Reply:
Hm, yeah it does. I'll change this. Good catch.